### PR TITLE
BOM-2789: Upgrade enterprise-catalog to Django 3.2

### PR DESCRIPTION
**Pull request by mzulqarnain1**
## Description

- Upgraded To Use Django 3.2
- Dropped Django 2.2 Support
- Ran Make Upgrade

## Ticket Link

Link to the associated ticket
[BOM-2789](https://openedx.atlassian.net/browse/BOM-2789)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -3,7 +3,7 @@
 -c constraints.txt
 -r github.in              # Forks and other dependencies not yet on PyPI
 
-Django>=2.2,<2.3          # Web application framework
+Django
```
Comment (awais786) [inline requirements/base.in:6]: Add some pin < 4.0 or 3.3

**Review by awais786 — APPROVED** [review]

File: requirements/base.in (hunk 1 above)
Comment (mzulqarnain1) [reply requirements/base.in:6]: @awais786 Pin is there in constraints.txt already. We don't need this.

**Review by iamsobanjaved — APPROVED** [review]

**Comment by natabene** [comment]
@macdiesel Could you please review this when you have a chance? I believe this is the final review for enterprise-catalog, after this we will just wait for prod deployment.

**Review by macdiesel — APPROVED** [review]

File: enterprise_catalog/urls.py [hunk 2]
```diff
@@ -19,25 +19,31 @@
...
-from rest_framework_swagger.views import get_swagger_view
+from edx_api_doc_tools import make_api_info, make_docs_urls
 
 from enterprise_catalog.apps.api import urls as api_urls
 from enterprise_catalog.apps.core import views as core_views
 
 
 admin.autodiscover()
 
+api_info = make_api_info(
```
Comment (macdiesel) [inline enterprise_catalog/urls.py:30]: I know you did not write this and that it's the paradigm of the library, but for the record I would like to state my dislike for abstractions that are called to create a object that just creates the object you need with the same exact data. 😿

**Review by macdiesel — APPROVED** [review]

**Comment by macdiesel** [comment]
I'm confused if the matrix in the ci file is:
```
matrix:
        python-version: ['py38']
        django-version: ['django32']
```

Why the CR is blocking on the django22 tests to run?

**Comment by mzulqarnain1** [comment]
> I'm confused if the matrix in the ci file is:
> 
> ```
> matrix:
>         python-version: ['py38']
>         django-version: ['django32']
> ```
> 
> Why the CR is blocking on the django22 tests to run?

Hey @macdiesel, Django22 is set as a mandatory check for merging PR. You need to disable it and make the django32 check compulsory. 

See https://github.com/edx/edx-search/pull/113#issuecomment-916896701 on how to disable the required check.

**Pull request merged**